### PR TITLE
Move myip to a separate package and add myip.Interface.

### PR DESCRIPTION
**Pull request by yegle**
This lays the ground to add more `myip.Interface` in the future.

### CODE REVIEW

File: myip/myip.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+// Interface defines the interface to get external IP address of running code.
+type Interface interface {
+	IP() (net.IP, error)
+}
+
+// Taobao implements Interface using Taobao's API
+type Taobao struct {
+	Interface
```
Comment (ayanamist) [inline myip/myip.go:33]: 这里为什么要继承一个接口？

File: myip/myip.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	m.ip = ip
+	m.Unlock()
+}
+
+// New return a new MyIP instance to get external IP address from given API.
+func New(api Interface) *MyIP {
+	return &MyIP{api: api}
+}
+
+func (m *MyIP) StartTaobaoIPLoop(cb func(oldIP, newIP net.IP)) {
```
Comment (ayanamist) [inline myip/myip.go:64]: 如果要解耦MyIP的话，这里是不是拆到Taobao里去会比较好？或者抽象出一个StartLoop方法？

File: main.go [hunk 3]
```diff
@@ -12,6 +12,7 @@ import (
 	"strings"
 	"time"
 
+	"github.com/ayanamist/gdns-go/myip"
```
Comment (ayanamist) [inline main.go:15]: 这里使用相对import会不会更好？
`myip "./myip"`

File: myip/myip.go (hunk 1 above)
Comment (yegle) [reply myip/myip.go:33]: Hmm this was by accident, removed.

File: myip/myip.go (hunk 2 above)
Comment (yegle) [reply myip/myip.go:64]: Right, it should be a `Loop` method later on, but that should be a separate PR (FYI, I'm going to implement a `AppEngine` struct that get external IP address via App Engine).

Comment (ayanamist) [reply myip/myip.go:64]: 嗯，你可以把那个实现放在这里一起提，来说明拆出单独的package的意义。如果走App Engine的接口的话，就需要用户配自己的地址了？配置文件是不是也得规划一下？还是对获取IP这块抽象出一个规范？记得不少网站可以通过`curl ifconfig.co`的方式直接获取到IP地址，支持这种方式，然后填endpoint进来？

File: main.go (hunk 3 above)
Comment (yegle) [reply main.go:15]: AFAIK, relative import are discouraged, but I can't seem to find the document that support it. I'm leaning toward using absolute import path everywhere.

Comment (ayanamist) [reply main.go:15]: 绝对import带来的一个问题是，当别人要fork时，就得把代码里的import全部改一遍，然后PR时容易把这个改动提回来，这个惨剧我在github上看到过两次了。

Comment (yegle) [reply main.go:15]: The _correct_ way to fork and send PR should be: go get the upstream repo, than change git remote to one's own forked repo.

But anyway I can't seem to make relative import to work. I get the following message trying to `go build` the package:

> can't load package: $GOROOT/src/github.com/ayanamist/gdns-go/main.go:15:2: local import "./myip" in non-local package

File: myip/myip.go (hunk 2 above)
Comment (yegle) [reply myip/myip.go:64]: Ah, yes, `ifconfig.co` support IPv6 too, which is more reliable than running one's own App Engine instance. Will work on additional code that uses `ifconfig.co`

**Pull request closed without merging**